src/BinaryParsers/PEBinary/ProgramDatabase/Pdb.cs

```diff
@@ -57,13 +58,15 @@ public sealed class Pdb : IDisposable, IDiaLoadCallback2
         /// <param name="pdbPath">The path to the pdb.</param>
         public Pdb(
             string pdbPath,
-            bool traceLoads = false)
+            bool traceLoads = false,
+            IDiaDataSource diaDataSource = null)
         {
             this.peOrPdbPath = pdbPath;
             this.loadTrace = traceLoads ? new StringBuilder() : null;
             this.globalScope = new Lazy<Symbol>(this.GetGlobalScope, LazyThreadSafetyMode.ExecutionAndPublication);
             this.writableSegmentIds = new Lazy<HashSet<uint>>(this.GenerateWritableSegmentSet);
             this.executableSectionContribCompilandIds = new Lazy<HashSet<uint>>(this.GenerateExecutableSectionContribIds);
+            this.diaDataSource = diaDataSource == null ? MsdiaComWrapper.GetDiaSource() : diaDataSource;

```

Comment:
>this.diaDataSource = diaDataSource == null ? MsdiaComWrapper.GetDiaSource() : diaDataSource; [](http://example.com/codeflow?start=12&length=92)

This change gives us the capacity to mock results from IDiaDataSource.

Reply:
nit: diaDataSource ?? MsDiaComWrapper.GetDiaSource()